modules/internal/DefaultRectangular.chpl

```diff
@@ -1186,25 +1355,14 @@ module DefaultRectangular {
           // If we detect that blk is never changed then then blk(rank) == 1.
           // Knowing this, we need not multiply the final ind(...) by anything.
           // This relies on us marking every function that modifies blk
-          if __primitive("optimize_array_blk_mult") {
```

Comment:
I think the comment above is now out of date, right? Also, can we now remove the "modifies blk" flag?